cmake/zephyr_module.cmake

```diff
@@ -69,6 +81,7 @@ if(WEST OR ZEPHYR_MODULES)
       # Match "<name>":"<path>" for each line of file, each corresponding to
       # one module. The use of quotes is required due to CMake not supporting
       # lazy regexes (it supports greedy only).
+      string(CONFIGURE ${module} module)
```

Comment:
Why is this needed now?

Reply:
because the `zephyr_module.py` will now place the CMake variable `${ZEPHYR_<MODULE_NAME>_CMAKE_DIR}` in the `zephyr_modules.txt` file when the `cmake-extern: True` is used in module.yml.

Doing it this way, allows a module root handler to set the correct value of  `${ZEPHYR_<MODULE_NAME>_CMAKE_DIR}` to point to the glue code.

When CMake code was always located in the module repo, then the path was known, and hence no need for variables.